doc/developer_guide/psyir.rst

```diff
@@ -37,72 +37,147 @@ The PSyclone Internal Representation (PSyIR)
 ############################################
 
 The PSyclone Internal Representation (PSyIR) is a language-independent
-AST that PSyclone uses to represent the PSy layer and the kernel
-code. The PSyIR can be constructed from scratch or produced from
-existing code using one of the front-ends (Readers) and it can be
-transformed back to a particular language using the back-ends (Writers)
-provided in PSyclone.
-
-DataTypes
-=========
-
-PSyIR DataTypes currently support Scalar and Array types via the
-``ScalarType`` and ``ArrayType`` classes. However, they are designed
-to be easily extended. For example, a structure could be created thus:
+Intermediate Representation that PSyclone uses to represent the PSy (Parallel
```

Comment:
My idea here was to first 'set the stage' with which are the design requirements of PSyIR (expressive, mutable, extensible, heterogeneous, normalized), so that the design decisions below can be linked to these requirements. But I agree it looks good as a more generic introduction to PSyIR.